app/models/invite.rb

```diff
@@ -1,16 +1,36 @@
 class Invite < ApplicationRecord
-  validate :user_to_must_exits, :user_cannot_send_invites_to_himself
-  validates :user_from_id, :user_to_id, numericality: true, presence: true
-  validates :account_id, uniqueness: { scope: :user_to_id, message: 'You cannot send invite twice' }
-  validates :status, presence: true, on: :update
   belongs_to :account
 
-  private
+  validate :user_cannot_send_invites_to_himself
+  validates :user_to_email, format: {
+    with: /\A([A-Z|a-z|0-9](\.|_){0,1})+[A-Z|a-z|0-9]\@([A-Z|a-z|0-9])+((\.){0,1}[A-Z|a-z|0-9]){2}\.[a-z]{2,3}\z/,
+    message: 'Email should be valid'
+  }
+  validates :user_from_id, :user_to_id, numericality: true, presence: true
+  # TODO: Sends invite to account once with that validation
+  # validates :account_id, uniqueness: { scope: :user_to_id, message: 'You cannot send invite twice' }
+  validates :status, inclusion: { in: [true, false] }, on: :update
+
+  def self.create_invite_with_rules(args)
+    invite = new(args[:invite_params])
+    rules = Rule.new(args[:rule_params])
 
-  def user_to_must_exits
-    User.exists?(user_to_id) || errors.add(:user_to_id, 'You can send invite just to existing user.')
+    invite.send_invite
+    invite.save && rules.save
   end
 
+  # TODO: Send with using background jobs.
+  def send_invite
```

Comment:
Try avoiding using model name word in this class methods.
Which option you like better?
1) `invite.send_invite`
2) `invite.send` or even `invite.send!`